Chore: 스터디 룸 참가 및 나가기 로직 수정

jungjinbeom opened the pull request:
### 스터디룸 참가 
- 스터디룸 참가 시 참가자 id key 수정
- 스터디룸 참가 시 참가자 ID 값 localStorage에 저장

### 스터디룸 나가기 
- 뒤로가기 여부 확인을 위해 historyAction 파라미터 추가
-  historyAction 값에 따라 뒤로가기 기능 block 여부 로직 추가
- 스터디룸 나가기 시 localStorage에 저장되어있는 참가자 ID 값 삭제
- 사용하지 않는 proceed 함수 삭제
- 스터디룸 나가기 모달 창 스터디룸 생성-> 스터디룸 나가기로 수정

jungjinbeom committed:
chore: 스터디룸 참가 API 수정
- 스터디룸 참가 시 참가자 id key 수정

jungjinbeom committed:
chore: 스터디룸 참가, 나가기 로직 수정
- 스터디룸 참가 시 참가자 ID 값 localStorage에 저장
- 뒤로가기 여부 확인을 위해 historyAction 파라미터 추가
-  historyAction 값에 따라 뒤로가기 기능 block 여부 로직 추가
- 스터디룸 나가기 시 localStorage에 저장되어있는 참가자 ID 값 삭제
- 사용하지 않는 proceed 함수 삭제

jungjinbeom committed:
chore: 스터디룸 나가기 모달 창 문구 오탈자 수정
- 스터디룸 나가기 모달 창 스터디룸 생성-> 스터디룸 나가기로 수정

Seung-wan commented on src/apis/study-room/join-study-room.ts:
```diff
@@ -5,13 +5,13 @@ interface JoinStudyRoomParams {
 
 interface JoinStudyRoomResponse {
-  userId: number;
+  participantId: number;
 }
 
 export const joinStudyRoom = async ({
   studyRoomId,
 }: JoinStudyRoomParams): Promise<JoinStudyRoomResponse> => {
   const { data } = await http.post(`/studyrooms/${studyRoomId}/participants`);
 
-  return { userId: data.id };
+  return { participantId: data.id };
```
JoinStudyRoomResponse라는 타입이라고 하면 저는 api의 response 타입이 그대로 매핑될 것 처럼 느껴지는데 개발자가 임의로 만든 데이터 형식을 표현하고 있어서 조금 어색하게 느껴지기도 하는데 혹시 어떠신가요?

jungjinbeom replied:
다른 API 들을 확인했을때는 api의 response 타입을 매핑에서 내려주는 이 API 만 이질적으로 참가자 ID 의 값을 내려주고 있네요 이부분은 컨벤션에 맞게 data를 내려주는 맞는거 같습니다

Seung-wan commented on src/pages/study-rooms/components/study-room-list.tsx:
```diff
@@ -18,8 +19,9 @@ const StudyRoomList = () => {
-      await joinStudyRoom({ studyRoomId });
+      const { participantId } = await joinStudyRoom({ studyRoomId });
 
+      setLocalStorage({ key: 'participantId', value: String(participantId) });
```
여기 인터페이스는 객체보단 key, value 이렇게 받는게 더 괜찮을 것 같아요

jungjinbeom replied:
LocalStorage 클래스 생성후 전체적으로 수정 계획에 있습니다

Seung-wan commented on src/pages/study-room/hooks/useExitStudyRoom.ts:
```diff
@@ -11,15 +11,19 @@ interface UseExitStudyRoom {
+    return currentLocation.pathname === nextLocation.pathname;
+  });
+
+  const { mutate: exitStudyRoomMutate } = useExitStudyRoomMutation({
+    close,
+    navigate,
+  });
```
여기서 close, navigate를 넘겨주기보다 onSuccess에서 실행할 callback을 넘겨주면 더 좋을 것 같아요

jungjinbeom committed:
chore: 스터디룸 참가 API 및 로직 수정
- API 리턴 값 수정
- API 리턴 값 수정에 따른 데이터 받는 방식 수정

jungjinbeom committed:
chore: 스터디룸 나기기 로직 수정
- 스터디룸 나가기 로직을 Mutation 함수 내부가 아닌 useExitStudyRoom 커스텀 훅에 선언 후 Mutation 함수에 넘겨주는 방식으로 로직 수정

Seung-wan commented on src/pages/study-room/hooks/useExitStudyRoom.ts:
```diff
@@ -11,15 +12,23 @@ interface UseExitStudyRoom {
 
-  const exitStudyRoom = () => {
-    if (blockState !== 'blocked') return;
+  const { mutate: exitStudyRoomMutate } = useExitStudyRoomMutation({ exitStudyRoomAction });
```
이렇게 표현되어야 내부 동작을 추측할 수 있을 것 같아요
```suggestion
  const { mutate: exitStudyRoomMutate } = useExitStudyRoomMutation({ onSuccess: exitStudyRoomAction });
```
